app/views/settings/memories/_memory.html.erb

```diff
@@ -1,4 +1,20 @@
-<li class="list-disc py-1">
-  <%= memory.detail %>
-  <%= link_to_source(memory).html_safe %>
-</li>
+<tr class="py-1 <%= cycle('bg-gray-100 dark:bg-gray-600', 'bg-white dark:bg-gray-900') %>">
+  <td>
+    <%= memory.detail %>
+    <%= link_to_source(memory).html_safe if memory.message.present? %>
```

Comment:
Looking at the model, I don't think the message can ever be blank but it doesn't hurt to be cautious

Reply:
If you delete it in the UI I think the memory remains.

Reply:
Ohhh, hmm, it looks like message is missing a has_one (or many?) memories so there is no delete propagation. Ack.

BTW, I finally fixed the test that I added and kept breaking. :) This PR looks great, thanks so much for teeing it up! If you want to do another PR on the memories association that would be great, but no obligation at all. I'll take whatever you're feeling motivated to work on.

Reply:
I think it would be annoying if the delete propagated.  Everything works great now and ChatGPT doesn't delete memories automatically either. 
I might do the thing about showing the tool call in the UI so the user knows what's happening.  I was going to do 'web search' but that seems pretty hard!